tests/babel.test.ts

```diff
@@ -0,0 +1,180 @@
+        `,
+            output: `
+            import { Computed } from '@another/package';
+            const fn = () => (
+                <Computed>
+                    <div>hi</div>
+                </Computed>
+            );
+        `,
+        },
+        'does not change Show without @legendapp/state/react': {
+            code: `
+            import { Show } from '@another/package';
+            const fn = () => <Show if={true}><div>hi</div></Show>;
+        `,
+            output: `
+            import { Show } from '@another/package';
+            const fn = () => (
+                <Show if={true}>
+                    <div>hi</div>
+                </Show>
+            );
+        `,
+        },
+        'does not change if expression has a ArrowFunctionExpression': `
+            import { Computed } from '@legendapp/state/react';
+            function Component() {
+                return (
+                    <Computed>
+                        {() =>
+                            state$.messages.map((message) => (
+                                <div key={message.id}>
+                                    {message.text} {localVar}
+                                </div>
+                            ))
+                        }
+                    </Computed>
+                );
+            }
+        `,
+        'does not change if expression has a FunctionExpression': `
+            import { Memo } from '@legendapp/state/react';
+            function C() {
+                return (
+                    <Memo>
+                        {function a() {
+                            state$.messages.map((message) => (
+                                <div key={message.id}>
+                                    {message.text} {localVar}
+                                </div>
+                            ));
+                        }}
+                    </Memo>
+                );
+            }
+        `,
+        'does not change if expression is Observable (MemberExpression)': `
+            import { Memo } from '@legendapp/state/react';
+            function C() {
+                return <Memo>{state$.count}</Memo>;
+            }
+        `,
+        'does not change if expression is Observable (Function Identifier)': `
+            import { Memo } from '@legendapp/state/react';
+            function C() {
+                return <Memo>{functionCall}</Memo>;
+            }
+        `,
+        'does not change for ConditionalExpression': `
+            import { Memo } from '@legendapp/state/react';
+            function C() {
+                return <Memo>{true ? state$.count : state$.count}</Memo>;
+            }
+        `,
+        // it could do this, since it doesn't return a function
+        'does not change for ConditionalExpression without Observable': `
+            import { Memo } from '@legendapp/state/react';
+            function C() {
+                return <Memo>{true ? 'hi' : 'bye'}</Memo>;
```

Comment:
The plugin could also transform expressions that don't return a function, but I don't know how easy that is to figure out, or it's too much magic already?

I had stumbled into this when testing, where I tried to do:

```tsx
<Memo>
  {store$.legend.get() ? 'useState' : 'Legend'}
</Memo>
// which I thought would become
<Memo>
  {() => (store$.legend.get() ? 'useState' : 'Legend')}
</Memo>
```

But this plugin only checks against `JSXText` meaning it needs to wrap in a `div` or something. The below works fine.

```tsx
<Memo>
  <div>{store$.legend.get() ? 'useState' : 'Legend'}</div>
</Memo>
```

Not sure what a user expects, maybe it's probably more important that the user knows what the plugin is actually doing, which is wrap the function call, given certain criteria.

Other issue would be that the wrap happens when it doesn't need to: the returned result of an expression already is a function to be called

```tsx
<Memo>
  <div>{store$.legend.get() ? fn : fn2}</div>
</Memo>
<Computed>
{state$.count}
</Computed>
```

Since a simple check isn't enough (old commit) https://github.com/LegendApp/legend-state/commit/da7eebb4860053dc8d64be1a2e913a4e4685f912

See [docs](https://legendapp.com/open-source/state/react/fine-grained-reactivity/#optionally-add-the-babel-plugin)